twx

```diff
@@ -349,51 +327,131 @@ execute_code() {
+                    </ServiceDefinition>
+                    </ServiceDefinitions>
+                    <ServiceImplementations>
+                    <ServiceImplementation description=\"\" handlerName=\"Script\" name=\"Run\">
+                        <ConfigurationTables>
+                            <ConfigurationTable dataShapeName=\"\" description=\"\" isMultiRow=\"false\" name=\"Script\" ordinal=\"0\">
+                                <DataShape>
+                                <FieldDefinitions>
+                                    <FieldDefinition baseType=\"STRING\" description=\"code\" name=\"code\" ordinal=\"0\" />
+                                </FieldDefinitions>
+                                </DataShape>
+                                <Rows>
+                                <Row>
+                                    <code><![CDATA[$eval_code]]></code>
+                                </Row>
+                                </Rows>
+                            </ConfigurationTable>
+                        </ConfigurationTables>
+                    </ServiceImplementation>
+                    </ServiceImplementations>
+                </ThingShape>
+            </Thing>
+        </Things>
+    </Entities>"
+
+    echo "$xml_template" >"$temp_file_name"
+
+    import_xml=$(import "$temp_file_name")
     exit_code=$?
 
-    rm -r "$UPDATE_FILE"
+    rm -r "$temp_file_name"
 
     if [ $exit_code -eq 2 ]; then
         echo "Error importing XML with .js file $js_file_name"
     else
         PARAM_ARRAY=()
         if [ $# -eq 0 ]; then
-            CALL_JS_SERVICE=$(call_service "Things/$TEMP_NAME/Run")
+            call_js_service=$(call_service "Things/$temp_entity_name/Run")
             exit_code=$?
         else
             for i in "$@"; do
                 PARAM_ARRAY+=("$i")
             done
 
-            CALL_JS_SERVICE=$(call_service "Things/$TEMP_NAME/Run" "${PARAM_ARRAY[@]}")
+            call_js_service=$(call_service "Things/$temp_entity_name/Run" "${PARAM_ARRAY[@]}")
             exit_code=$?
         fi
 
         if [ $exit_code -ne 5 ]; then
-            echo "$CALL_JS_SERVICE"
+            echo "$call_js_service"
         else
-            echo "Error run .js file $js_file_name. Response: $CALL_JS_SERVICE"
+            echo "Error run .js file $js_file_name. Response: $call_js_service"
         fi
     fi
 
-    call_service "Resources/EntityServices/DeleteThing" "-pname=$TEMP_NAME" > /dev/null
+    call_service "Resources/EntityServices/DeleteThing" "-pname=$temp_entity_name" > /dev/null
+
+    return ${exit_code}
+}
+
+build_ext() {
+    local exit_code=0
+    local error_exit_code=8
+
+    if [$# -lt 2]; then 
+        exit_code=$error_exit_code
+        return ${exit_code}
+    fi
+
+    local ext_src_dir="$1"
+    local ext_name="$2"
+    local ext_version="$3"
+
+    if [ -z "$3" ]; then
+        ext_version="1.0.0"
+    fi
+
+    local build_dir="./build"
+    local temp_build_dir="./build-tmp"  
```

Comment:
Better use the same `mktemp` mechanism as we do for running code above.

Reply:
Done.